product-service/product-service/src/models/product.model.ts

```diff
@@ -1,5 +1,5 @@
 export interface ProductModel {
-    id: string;
+    id?: string;
```

Comment:
To have the flexibility you can create something like ProductData structure which doesn't have any data layer related stuff and extend ProductModel from it by just adding id :)

Reply:
thanks